src/vs/platform/editor/common/editor.ts

```diff
@@ -140,4 +140,9 @@ export interface ITextEditorOptions extends IEditorOptions {
 	 * Option to scroll vertically or horizontally as necessary and reveal a range centered vertically only if it lies outside the viewport.
 	 */
 	revealInCenterIfOutsideViewport?: boolean;
+
+	/**
+	 * An optional contents string that needs to be set when opening document.
+	 */
+	contents?: string;
```

Comment:
We cannot change the vscode APIs here. This will get squashed in any merge. For any change like this we'd need VSCode to accept the request and these are very stable APIs, so they likely have reasons not to do this.

Reply:
Ok, I will rework on this not to modify the vscode API.